business_requirement_deliverable/models/business.py

```diff
@@ -0,0 +1,272 @@
+# -*- coding: utf-8 -*-
+# © 2016 Elico Corp (https://www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models
+from openerp.exceptions import ValidationError, Warning
+from openerp.tools.translate import _
+
+
+class BusinessRequirementResource(models.Model):
+    _name = "business.requirement.resource"
+    _description = "Business Requirement Resource"
+
+    sequence = fields.Integer('Sequence')
+    description = fields.Char('Description', required=True)
+    product_id = fields.Many2one(
+        comodel_name='product.product',
+        string='Product',
+        required=False
+    )
+    uom_id = fields.Many2one(
+        comodel_name='product.uom',
+        string='UoM',
+        required=True
+    )
+    qty = fields.Float(
+        string='Quantity',
+        default=1,
+    )
+    resource_type = fields.Selection(
+        selection=[('task', 'Task'), ('procurement', 'Procurement')],
+        string='Type',
+        required=True,
+        default='task'
+    )
+    user_id = fields.Many2one(
+        comodel_name='res.users',
+        string='Assign To',
+        ondelete='set null'
+    )
+    business_requirement_deliverable_id = fields.Many2one(
+        comodel_name='business.requirement.deliverable',
+        string='Business Requirement Deliverable',
+        ondelete='cascade'
+    )
+
+    @api.multi
+    @api.onchange('product_id')
+    def product_id_change(self):
+        description = ''
+        uom_id = False
+        product = self.product_id
+        if product:
+            description = product.name_get()[0][1]
+            uom_id = product.uom_id.id
+        if product.description_sale:
+            description += '\n' + product.description_sale
+        self.description = description
+        self.uom_id = uom_id
+
+    @api.onchange('resource_type')
+    def resource_type_change(self):
+        if self.resource_type == 'procurement':
+            self.user_id = False
+
+    @api.multi
+    @api.constrains('resource_type', 'uom_id')
+    def _check_description(self):
+        if self.resource_type == 'task'\
+                and self.uom_id.category_id.name != 'Working Time':
```

Comment:
Can't we use the more precise XML id instead?

    self.uom_id.category_id != self.ref('<working_time_xml_id'>'):